org.eclipse.january.examples/src/org/eclipse/january/examples/dataset/AxesMetadataExample.java

```diff
@@ -0,0 +1,290 @@
+package org.eclipse.january.examples.dataset;
+
+import java.util.Arrays;
+
+import org.eclipse.january.dataset.CompoundDataset;
+import org.eclipse.january.dataset.Dataset;
+import org.eclipse.january.dataset.DatasetFactory;
+import org.eclipse.january.dataset.IndexIterator;
+import org.eclipse.january.dataset.LongDataset;
+import org.eclipse.january.dataset.Maths;
+import org.eclipse.january.metadata.AxesMetadata;
+import org.eclipse.january.metadata.internal.AxesMetadataImpl;
+import org.eclipse.uomo.units.SI;
+import org.unitsofmeasurement.unit.Unit;
+
+public class AxesMetadataExample {
+
+	public static void main(String[] args) {
+		testDataset();
+		testCompoundDataset();
+		testUnitsAwareOperations();
+	}
+
+	private static class NewMaths extends Maths
+	{
```

Comment:
@dinkoivanov - I've had a go at wrapping the existing Maths operations.  The strategy is able to populate the results with correct UoM metadata.  

I haven't considered observing the AxisMetadata object.